helios-tools/src/main/java/com/spotify/helios/cli/command/HostListCommand.java

```diff
@@ -147,11 +157,13 @@ int run(final Namespace options, final HeliosClient client, final PrintStream ou
       try {
         final Map<String, HostStatus> hostStatuses = client.hostStatuses(hosts).get();
         for (final Entry<String, HostStatus> entry : hostStatuses.entrySet()) {
-          statuses.put(entry.getKey(), Futures.immediateFuture(entry.getValue()));
+          if (statusFilter == null || entry.getValue().getStatus().equals(statusFilter)) {
```

Comment:
Maybe just uber-paranoia but it becomes NPE-proof if you do `statusFilter.equals( ... )` instead.